lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -64,23 +65,39 @@ local function adjustMovementSpeed(ply)
     local wepCount = 0
     
     -- count weapons
-    for k, weapon in pairs(weapons) do
+    for _, weapon in pairs( weapons ) do
         if nonEffectedWeapons[weapon:GetClass()] == nil then
-            wepCount =  wepCount+1
+            wepCount =  wepCount + 1
         end
     end
+    wepCount = math.max(wepCount + wepNum, 0)
```

Comment:
the wepnum value is passed then added to the count because the hooks onEquip and onDrop are called before any actual change is made to the players weapons.
when dropping a significant weapon you pass -1
when equipping you pass 1

Reply:
i added max because i was concerned that it might mess up the math
but  "if weaponCount < 1 then return 1 end" fixes that so it can probably be removed